src/Core/Checkout/Cart/Error/Error.php

```diff
@@ -25,6 +25,30 @@ abstract class Error extends \Exception implements \JsonSerializable
 
     final public const LEVEL_ERROR = 20;
 
+    /**
+     * The trace has to be cleaned up to remove service references that are not serializable.
+     *
+     * @return array<string, mixed>
+     */
+    public function __serialize(): array
+    {
+        $ref = new \ReflectionClass($this);
+
+        $data = [];
+        foreach ($ref->getProperties() as $property) {
+            $data[$property->getName()] = $property->getValue($this);
+        }
+
+        $traces = $this->getTrace();
+        foreach ($traces as &$trace) {
+            $trace['args'] = [];
+        }
+
+        $data['trace'] = $traces;
```

Comment:
Serializing private properties is good, but I would like to exclude the serialization of the trace. It will just clutter up the database with information that nobody is really interested in :thinking: especially since cart data can become stale and sums up over time